src/core/tokenizer.ts

```diff
@@ -39,29 +44,67 @@ const FIXED_TOKENS = [
   "<!--",
   "-->",
 ];
-const REPEATABLE_SYMBOLS_REGEX = REPEATABLE_SYMBOLS.map(escapeRegExp)
-  .map((s) => `${s}+`)
-  .join("|");
-const FIXED_TOKENS_REGEX = FIXED_TOKENS.map(escapeRegExp).join("|");
+
 const IDENTIFIERS_REGEX = "[\\p{L}_0-9]+";
 const SINGLE_SYMBOLS_REGEX = "[^\\s\\w]";
 const NUMBERS_REGEX = "(?<=[^.\\d]|^)\\d+\\.\\d+(?=[^.\\d]|$)"; // (not-dot/digit digits dot digits not-dot/digit)
 
-const REGEX = [
-  FIXED_TOKENS_REGEX,
-  NUMBERS_REGEX,
-  IDENTIFIERS_REGEX,
-  REPEATABLE_SYMBOLS_REGEX,
-  SINGLE_SYMBOLS_REGEX,
-].join("|");
+const defaultLanguageTokenizerSetting: LanguageTokenizerComponents = {
+  fixedTokens: FIXED_TOKENS,
+  repeatableSymbols: REPEATABLE_SYMBOLS,
+  identifiersRegex: IDENTIFIERS_REGEX,
+  numbersRegex: NUMBERS_REGEX,
+  singleSymbolsRegex: SINGLE_SYMBOLS_REGEX,
+};
+
+const defaultTokenMatcher = generateTokenMatcher();
+
+function generateTokenMatcher(
+  languageSetting: LanguageTokenizerComponents = defaultLanguageTokenizerSetting
+): RegExp {
+  const tokenizerDefinition = Object.assign(
+    { ...defaultLanguageTokenizerSetting },
+    languageSetting
+  );
+
+  const repeatableSymbolsRegex = tokenizerDefinition.repeatableSymbols
+    .map(escapeRegExp)
+    .map((s) => `${s}+`)
+    .join("|");
 
-export const TOKEN_MATCHER = new RegExp(REGEX, "gu");
+  const fixedTokensRegex = tokenizerDefinition.fixedTokens
+    .map(escapeRegExp)
+    .join("|");
+
+  // Order matters here.
+  const regex = [
+    fixedTokensRegex,
+    tokenizerDefinition.numbersRegex,
+    tokenizerDefinition.identifiersRegex,
+    repeatableSymbolsRegex,
+    tokenizerDefinition.singleSymbolsRegex,
+  ].join("|");
+  return new RegExp(regex, "gu");
+}
+
+const tokenMatchersForLanguage: Partial<Record<SupportedLanguageId, RegExp>> =
```

Comment:
nit: I'd prob store `{ css, scss }` in a variable first, so that the dev doesn't need to look at this complex statement to add a new one